src/FwdState.cc

```diff
@@ -815,6 +816,8 @@ FwdState::noteConnection(HappyConnOpener::Answer &answer)
     if (answer.reused) {
         syncWithServerConn(answer.conn, request->url.host(), answer.reused);
         return dispatch();
+    } else {
+        serverConn = answer.conn.clone();
```

Comment:
I was hoping that using this kind of "late" cloning would remove the need to give write access to the in-destinations Connection object _and_ would remove the first cloning call in PeerDestinations.cc, but we really cannot remove the first cloning when FwdState has only one Connection storage because FwdState needs write access to the store Connection.

With the cloneDestinationDetails() intact, I see no reason to clone here. Can we remove this clone() call and the clone() method with it? The in-destinations Connection is going to be protected by that cloneDestinationDetails() call in ResolvedPeers::extractFound(). The resolved peer connection recipients are free to modify the stored Connection.

Reply:
> I see no reason to clone here

AFAIK, after 25b0ce4 FwdState should not possess an opened (but half-baked) server connection before it has passed through all required 'steps' and FwdState::syncWithServerConn() is called. There are many places in FwdState, performing actions depending on Comm::IsConnOpen(serverConn), e.g.,:

```
if (Comm::IsConnOpen(serverConn))
    unregister(serverConn);
```

I would not change/violate this logic - that is why cloning is required here.

Reply:
Thank you, I now see what you are trying to accomplish, and I agree that it is important to keep serverConn unopened until we are ready to call syncWithServerConn(). There may be more related changes needed, but I consider _this_ change request addressed.